File: atcoder/abc132B_ordinary_number.ml
(* Vicfred
 * https://atcoder.jp/contests/abc132/tasks/abc132_b
 * implementation
 * *)
let main = Scanf.scanf "%d\n" @@ fun n ->
    let p = Array.init n @@ fun _ -> Scanf.scanf " %d" ( + ) 0 in
    Array.init (n - 2) (fun i -> p.(i), p.(i + 1), p.(i + 2))
    |> Array.fold_left (fun v (x, y, z) -> if (x < y && y < z) || (x > y && y > z) then v + 1 else v) 0
    |> Printf.printf "%d\n"

File: atcoder/abc132B_ordinary_number.py
# vicfred
# https://atcoder.jp/contests/abc132/tasks/abc132_b
# implementation
n = int(input())
p = list(map(int,input().split()))

answer = 0
for i in range(1, n-1):
    if p[i-1] < p[i] < p[i+1] or p[i-1] > p[i] > p[i+1]:
        answer += 1

print(answer)

